### ISSUE TITLE
Remove dates from the per-file copyright header

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#4857]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)| @exarkun reported|
|-|-|
|Trac ID|trac#4857|
|Type|task|
|Created|2011-02-09 13:27:24Z|
|Branch|https://github.com/twisted/twisted/tree/consolidate-copyright-dates-4857|

Currently we include this header at the top of every file in Twisted:

```
# Copyright (c) 2001-2010 Twisted Matrix Laboratories.
# See LICENSE for details.
```

with only the dates varying, depending (sometimes) on when the file was introduced and when it was last modified.

The primary outcome of this is that most reviews start with the comment:

  1. Please update the dates in the copyright header in modified files

This is strictly a waste of time for developers and reviewers.

The copyright header on all files should be changed to:

```
# Copyright (c) Twisted Matrix Laboratories.
# See LICENSE for details.
```

We should maintain a date _only_ in the `LICENSE` file, and update it only at release times.

Various wiki and developer howtos should be updated to remove mentions of copyright date updates, as well.

Attachments:

* [rewrite-copyright-header.py](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/abb/abb2e1a13ed77d8474c3b10faa63c9332574605f/bd5f5f0ea9018f6439b1322efa6a313c6592e6d8.py) (875 bytes) - added by exarkun on 2011-02-13 15:54:00Z - tool which did all of the header rewriting

<details><summary>Searchable metadata</summary>

```
trac-id__4857 4857
type__task task
reporter__exarkun exarkun
priority__normal normal
milestone__ 
branch__branches_consolidate_copyright_dates_4857 branches/consolidate-copyright-dates-4857
branch_author__exarkun exarkun
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1297258044000000 1297258044000000
changetime__1306083655000000 1306083655000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_1"></a>|@exarkun commented|
|-|-|

(In [30713]) Branching to 'consolidate-copyright-dates-4857'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun commented|
|-|-|

(In [30714]) Change the copyright headers with the rewrite-copyright-header.py script

refs [#4857](https://github.com/twisted/twisted/issues/4857)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun commented|
|-|-|

The branch has rewritten headers.  All changes were made by the tool attached to the ticket.

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_4"></a>|@glyph set owner to @exarkun|
|-|-|

Yes.  Please.  Let's land this and centralize all the copyright brouhaha.  If anyone cares about dates of individual files, version control will always be there to tell them.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_6"></a>|@exarkun set status to `closed`|
|-|-|

(In [30752]) Rewrite the copyright headers to exclude date information.

Author: exarkun
Reviewer: glyph
Fixes: [#4857](https://github.com/twisted/twisted/issues/4857)

To avoid the need to perpetually update copyright dates in each file in Twisted,
remove the dates from most files and just leave them in the LICENSE file.

As a side effect, some files also have had a trailing newline added where it was
missing before.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_7"></a>|@exarkun commented|
|-|-|

(In [31931]) Merge copyright-policy-5112

Author: exarkun
Reviewer: ivank
Fixes: [#5112](https://github.com/twisted/twisted/issues/5112)

Remove the recommendation to include the year in copyright headers, matching
what we currently actually want to do with respect to these.  See [#4857](https://github.com/twisted/twisted/issues/4857).
